include/Spectra/GenEigsBase.h

```diff
@@ -425,6 +441,8 @@ class GenEigsBase
         for (i = 0; i < maxit; i++)
         {
             nconv = num_converged(tol);
+            if (m_logger)
+                m_logger->iteration_log(i, nconv, m_ncv, m_ritz_val.head(m_nev), m_resid, m_ritz_conv);
```

Comment:
I think it would be nicer to have an iteration_data structure because if we want to log more things this function signature would have to change everywhere, of course then we would maybe have to take copies.

Reply:
Sure this makes sense. I will push some commits with this change. Even if we make copies, the size of the objects shouldn't be overly large so this will probably be ok.